### PR TITLE
Add GitHub issue forms

### PR DESCRIPTION
**Pull request by jtoar**
This PR adds three GitHub issue forms, one for bug reports, documentation, and feature requests. Now when users open an issue, they'll choose from one of those templates.

The motivation for this is that we want...

1. issues to be more structured, (specifically, bug reports)
2. to be more upfront about our process—i.e., what happens after you open an issue

I used GitHub issue forms when opening an issue in the yarn/berry repo and thought the process was much better than a blank textarea.

Note that the documentation and feature request templates aren't really that different from a blank GitHub issue, but I thought providing a blurb about expectations was worthwhile enough to warrant templates for them. We can add to these templates later if we find that we want more information.

**See it in action**

You can see how this would look like on my fork of redwood here: https://github.com/orgtoar/redwood-test. Try opening an issue. What's also nice is that GitHub renders these files: https://github.com/orgtoar/redwood-test/blob/main/.github/ISSUE_TEMPLATE/bug-report.yml.

**Tape**

https://user-images.githubusercontent.com/32992335/174222888-236fee25-7a03-4a82-b440-334f0d253357.mov

### CODE REVIEW

**Comment by jtoar** [comment]
@dac09 this is ready to go pending a link to docs or a demo of making a reproduction in Gitpod. Let me know if you need help with that!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,34 +16,6 @@ Love Redwood and want to get involved? You’re in the right place!
 
 _Before interacting with the Redwood community, please read and understand our [Code of Conduct](https://github.com/redwoodjs/redwood/blob/main/CODE_OF_CONDUCT.md)._
 
-**Table of Contents**
```
Comment (dac09) [inline CONTRIBUTING.md:19]: @jtoar removed this because Github now gives you a ToC anyway

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+body:
+  - type: textarea
+    attributes:
+      label: What's not working?
+      description: >
+        Describe the bug here.
+        The more detail you provide, the better,
+        but try to start with a summary to frame the issue.
+      placeholder: >
+        The dev server doesn't seem to be working.
```
Comment (dac09) [inline .github/ISSUE_TEMPLATE/bug-report.yml:16]: This stuff really is too wordy, no? 

I definitely would take out the suggestion about bumping webpack, because guesses like this unless linked with a specific issue are probably more harmful than helpful

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        If we bump the version Redwood is using, it may resolve this.
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: How do we reproduce the bug?
+      description: >
+        What commands do we need to run?
+        Any packages do we need to install?
```
Comment (jtoar) [inline .github/ISSUE_TEMPLATE/bug-report.yml:34]:
```suggestion
        Are there any packages that we need to install?
```

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 2 above)
Comment (jtoar) [reply .github/ISSUE_TEMPLATE/bug-report.yml:16]: The motivation here was to provide people opening issues with an example of what a good one looks like, but instead of including it here, we can just link out to it after we have a good one written up with the new template.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -56,6 +28,23 @@ Chances are that you'll have more than a few VS Codes open when you're contribut
 
 ![image](https://user-images.githubusercontent.com/32992335/130697522-313317f8-21e5-4f71-8b8e-9690dbad412a.png)
 
+
+## Creating a reproduction to include with issues
+Including a reproduction, either as a series of steps, a public Github repo with the problem you are facing or a Gitpod snapshot will definitely let us help you faster!
```
Comment (jtoar) [inline CONTRIBUTING.md:33]:
```suggestion
## Creating a Reproduction to Include with Issues
Are you about to open an issue? Including a reproduction, either as a series of steps, as a public GitHub repo, or as a Gitpod snapshot, will definitely let us help you faster!
```

**Review by jtoar — COMMENTED** [review]
Thanks for the video @dac09! It's even got a few edits in it—must've been a lot of work! Which is why I feel bad asking for a change. 😅 The content itself is great, it's just the silence from around 0:47 to around 1:00 is a little awkward. Is there anything we can do about that? I feel bad asking you to redo that part so I'm happy to try and come up with something if you're not up to it. Or we can propose that Rob do it at the next meeting. 😉

I've suggested some edits to CONTRIBUTING.md that I'm going to commit. I made them comments for traceability. Lastly, I'm going to move the new section cause it interrupts the flow of the document a bit.
